trieste/models/gpflow/architectures.py

```diff
@@ -0,0 +1,228 @@
+    (optionally) priors. We use :class:`~gpflow.kernels.Matern52` kernel in the model as we found
+    it to be effective in most settings. We assume inputs are normalised to the unit hypercube for
+    setting sensible initial parameters for the kernel.
+
+    We set priors for kernel hyperparameters by default in order to stabilize model fitting. We
+    found the priors below to be highly effective for objective functions defined over the unit
+    hypercube. For objective functions with different scaling, other priors will likely be more
+    appropriate. Using priors allows for using maximum a posteriori estimate of these kernel
+    parameters during model fitting.
+
+    :param data: Dataset from the initial design, used for estimating the variance of observations.
+    :param noise_free: If set to `True` Gaussian likelihood paramater is set to
+        non-trainable. By default set to `False`.
+    :param priors: If set to `True` (default) priors are set for kernel parameters.
+    :return: A :class:`~gpflow.models.GPR` model.
+    """
+    empirical_mean, empirical_variance, _ = _get_data_stats(data)
+
+    kernel = _get_kernel(empirical_variance, search_space, kernel_priors)
+    mean = gpflow.mean_functions.Constant(empirical_mean)
+
+    if likelihood_variance is None:
+        noise_variance = 0.1 * empirical_variance
+    else:
+        noise_variance = likelihood_variance
+
+    model = gpflow.models.GPR(data.astuple(), kernel, mean, noise_variance)
+
+    gpflow.set_trainable(model.likelihood, (not noise_free))
+
+    return model
+
+
+def build_sgpr(
+    data: Dataset,
+    search_space: Box,
+    kernel_priors: bool = True,
+    noise_free: bool = False,
+    likelihood_variance: Optional[float] = None,
+    num_inducing_points: Optional[int] = None,
+    trainable_inducing_points: bool = False,
+) -> SGPR:
+    empirical_mean, empirical_variance, _ = _get_data_stats(data)
+
+    kernel = _get_kernel(empirical_variance, search_space, kernel_priors)
+    mean = gpflow.mean_functions.Constant(empirical_mean)
+
+    inducing_points = _get_inducing_points(search_space, num_inducing_points)
+
+    if likelihood_variance is None:
+        noise_variance = 0.1 * empirical_variance
+    else:
+        noise_variance = likelihood_variance
+
+    model = gpflow.models.SGPR(
+        data.astuple(), kernel, inducing_points, mean_function=mean, noise_variance=noise_variance
+    )
+
+    gpflow.set_trainable(model.likelihood, (not noise_free))
+    gpflow.set_trainable(model.inducing_variable, trainable_inducing_points)
+
+    return model
+
+
+def build_vgp(
+    data: Dataset,
+    search_space: Box,
+    classification: bool = False,
+    kernel_priors: bool = True,
+    noise_free: bool = False,
+    likelihood_variance: Optional[float] = None,
+) -> VGP:
+    empirical_mean, empirical_variance, _ = _get_data_stats(data)
+
+    if classification:
+        if noise_free:
+            empirical_variance = 100.0
```

Comment:
this is a good trick, but we should also apply it to the empirical_mean. I guess for classification the default mean should be zero.

Reply:
I would personally move this logic into the block below, i.e.

```
if classification and noise_free:
        model.kernel.variance.assign(100.0)
        gpflow.set_trainable(model.kernel.variance, False)
```
and leave the default empirical variance to 1.0

Reply:
You could also strip the prior (if one exists) from the kernel here, and simplify your `_get_kernel` function to not need specific noise_free logic; i.e. you contain all the special logic for this 'deterministic classification' case into one block (could be its own function if you wish)

Reply:
thanks, managed to simplify it